grtc/app.js

```diff
@@ -0,0 +1,471 @@
+'use strict';
+
+import Peer from 'simple-peer';
+import uuidv4 from 'uuid/v4';
+import uuidv1 from 'uuid/v1';
+import { EventEmitter } from 'events';
+
+import crypto from './crypto';
+
+/**
+ * Signal class that uses promise not event emitters and can be used standalone.
+ * getSignal and updateSignal lets you update and fetch data regarding the shared key.
+ */
+class Signal {
+	/**
+	 * @param {string} url url to which signal should go.
+	 * @param {string} grtcID global id shared by url
+	 * @param {string} peerID global peerID that maps to username.
+	 * @param {string} peerName global username that maps to peerID.
+	 * @param {object} signalID signal generated by peer
+	 * url is base url of page.
+	 * grtcID is collaborate param from url.
+	 * paramID is changing peerID on refresh or every new grtc instance.
+	 * signalID is peer signal used to traverse and connect P2P.
+	 */
+	constructor( url, grtcID, peerID, peerName, signalID ) {
+		const self = this;
+		self.url = url;
+		self.grtcID = grtcID;
+		self.peerID = peerID;
+		self.signalID = signalID;
+		self.peerName = peerName;
+	}
+
+	/**
+	 * Clear the key forcefully in kv.
+	 * @return {promise} promise object
+	 */
+	clearSignal() {
+		const self = this;
+		return new Promise( ( resolve, reject ) => {
+			const data = { peerID: self.peerID, peerName: self.peerName, type: 'initial', signal: self.signalID };
+			jQuery.post( self.url + '/remove', { [ self.grtcID ]: window.btoa( JSON.stringify( data ) ) }, ( resp ) => {
```

Comment:
jQuery is not defined as a dependency for this script (i.e. incidental that another plugin depends on jQuery, but cannot rely on this). In any case, we might want to consider if we can do without it (maybe `fetch`, which is already promise based?)

Reply:
My initial iteration uses `window.fetch`. We can update later id we decide on some different approach.